tests/dialects/test_dialect.py

```diff
@@ -1444,6 +1444,46 @@ def test_cross_join(self):
             },
         )
 
+        # UNNEST without column alias
+        self.validate_all(
+            "SELECT * FROM x CROSS JOIN UNNEST(y) AS t",
+            write={
+                "presto": "SELECT * FROM x CROSS JOIN UNNEST(y) AS t",
+                "spark": UnsupportedError,
+                "databricks": UnsupportedError,
+            },
+        )
+
+        # UNNEST STRUCT Object into multiple columns, using single alias
+        self.validate_all(
+            "SELECT a, b FROM x CROSS JOIN UNNEST(y) AS t (a, b)",
+            write={
+                "presto": "SELECT a, b FROM x CROSS JOIN UNNEST(y) AS t(a, b)",
+                "spark": "SELECT a, b FROM x LATERAL VIEW EXPLODE(y) t AS a, b",
+                "hive": "SELECT a, b FROM x LATERAL VIEW EXPLODE(y) t AS a",
```

Comment:
Why is Hive not following Spark here with the amount of column aliases?

Reply:
It's happening due to this previous [change](https://github.com/tobymao/sqlglot/blob/main/sqlglot/transforms.py#L379) for `hive`, where 
`unnest_using_arrays_zip` is being sent as false by default to prevent multiple alias.

`columns=alias_cols if unnest_using_arrays_zip else [column],  # type: ignore`

Reply:
Okay, I tested few more scenario on `hive` for each type:  array, map and struct, 
and hive does not support exploding struct.
`Error while compiling statement: FAILED: UDFArgumentException explode() takes an array or a map as a parameter`

Query:
```
WITH cte_x AS ( SELECT 1 AS id, STRUCT('value1', 123, TRUE) AS y UNION ALL SELECT 2 AS id, STRUCT('value2', 456, FALSE) AS y ) SELECT * FROM cte_x LATERAL VIEW EXPLODE(y) t AS a,b,c
```
Array requires single alias
```
WITH cte_x AS ( SELECT 1 AS id, ARRAY("value1", "value2", "value3") AS y UNION ALL SELECT 2 AS id, ARRAY("value4", "value5", "value6") AS y ) SELECT * FROM cte_x LATERAL VIEW EXPLODE(y) t AS a
```

Map requires 2

```
WITH cte_x AS ( SELECT 1 AS id, MAP("key1", "value1", "key2", "value2") AS y UNION ALL SELECT 2 AS id, MAP("key3", "value3", "key4", "value4") AS y ) SELECT * FROM cte_x LATERAL VIEW EXPLODE(y) t AS a,b
```

which is essentially the same behaviour as spark except it does not support the struct. I've fixed the code and testcases to support correct exception handling.